src/php/Analyzer/AnalyzeArray.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Phel\Analyzer;
+
+use Phel\Analyzer;
+use Phel\Ast\ArrayNode;
+use Phel\Lang\PhelArray;
+use Phel\NodeEnvironment;
+
+final class AnalyzeArray
+{
+    private Analyzer $analyzer;
+
+    public function __construct(Analyzer $analyzer)
```

Comment:
You can also make a `BaseAnalyzer` class that defines the constructor and the `$analyzer` property. That would mean less duplicated code.

Reply:
This "duplicated code" isn't really hurting at all, in my opinion. Actually, it makes every class independent from each other. They can be constructed in different ways. Look for example `AnalyzeLiteral`, it doesn't need the `Analyzer` at all.